app/src/main/java/org/akvo/flow/presentation/settings/passcode/PassCodeDialog.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2018 Stichting Akvo (Akvo Foundation)
+ *
+ * This file is part of Akvo Flow.
+ *
+ * Akvo Flow is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Akvo Flow is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Akvo Flow.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.akvo.flow.presentation.settings.passcode;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.app.DialogFragment;
+import android.support.v4.app.FragmentActivity;
+import android.support.v4.app.FragmentManager;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.EditText;
+
+import org.akvo.flow.R;
+import org.akvo.flow.util.ConstantUtil;
+
+public abstract class PassCodeDialog extends DialogFragment {
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        Context context = getContext();
+        AlertDialog.Builder builder = new AlertDialog.Builder(context);
+        View main = LayoutInflater.from(context).inflate(R.layout.pass_code_input_dialog, null);
+        builder.setTitle(R.string.authtitle);
+        builder.setMessage(R.string.authtext);
+        final EditText input = (EditText) main.findViewById(R.id.pass_code_et);
+        builder.setView(main);
+        builder.setPositiveButton(R.string.okbutton, new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                String val = input.getText().toString();
+                if (ConstantUtil.ADMIN_AUTH_CODE.equals(val)) {
+                    onPassCodeCorrect();
+                    if (dialog != null) {
+                        dialog.dismiss();
+                    }
+                } else {
+                    onPassCodeError();
+                    if (dialog != null) {
+                        dialog.dismiss();
+                    }
+                }
+            }
+        });
+
+        builder.setNegativeButton(R.string.cancelbutton,
+                new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        dismiss();
+                    }
+                });
```

Comment:
formatting seems a bit off here.

Reply:
Actually it is correct, I have tried formatting shortcuts and it correct